aws/cloud-formation-template.yml

```diff
@@ -1003,7 +1005,8 @@ Resources:
             setup:
               command: !Sub |
                 cat /etc/environment >> /etc/sysconfig/airflow
-
+                
+                FERNET_KEY=$(python -c "from cryptography.fernet import Fernet; FERNET_KEY = Fernet.generate_key().decode(); print(FERNET_KEY)")
```

Comment:
Nice work figuring out the injection! ;-)

But I'm afraid doing this for the Fernet Key makes the deployment become stateful (all other variables are deterministic). If try to update the stack, this piece of code will run again and generate a different key and we will lose the old key - also losing everything encrypted with it.

We must either receive the Fernet Key as parameter (greatly complicating the stack setup for new users) or find a parametric deterministic way to generate that key :-/

Reply:
I'm not sure if it's even safe to generate the key safely in that way, but we could create a passphrase parameter and use that parameter as seed to generate 32 cryptographically secure random bytes and then urlsafe base64 encode them, serving as Fernet key.

Reply:
Another way is to generate it only if it hasn't been generated once, similar to how the `/mnt/efs` mount point is handled.

Reply:
> Nice work figuring out the injection! ;-)
> 
> But I'm afraid doing this for the Fernet Key makes the deployment become stateful (all other variables are deterministic). If try to update the stack, this piece of code will run again and generate a different key and we will lose the old key - also losing everything encrypted with it.
> 
> We must either receive the Fernet Key as parameter (greatly complicating the stack setup for new users) or find a parametric deterministic way to generate that key :-/

yeah, I didn't think in that possibility, you're right, but passing the fernet_key as parameter can be difficult for beginners, not everyone knows how to generate it

> Another way is to generate it only if it hasn't been generated before, similar to how the `/mnt/efs` mount point is handled.

This approach seems very promising.